SurfaceTopography/Container/IO.py

```diff
@@ -0,0 +1,61 @@
+#
+# Copyright 2021 Lars Pastewka
+#
+# ### MIT license
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+import yaml
+from zipfile import ZipFile
+
+from .. import read_topography
+from .SurfaceContainer import SurfaceContainer
+
+
+def read_container(fn):
+    """
+    Read all surface in a container file and associated metadata. The
+    container is a ZIP file with raw data files and a YAML file meta.yml
+    that contains all metadata not contained in the raw data files.
+    """
+
+    surfaces = []
+
+    with ZipFile(fn, 'r') as z:
+        meta = yaml.load(z.open('meta.yml'))
+
+        topographies = []
+        for surf_meta in meta['surfaces']:
+            for topo_meta in surf_meta['topographies']:
+                t = read_topography(
+                    z.open(topo_meta['datafile']),
```

Comment:
I suggest to make the name of the field with the filename, here 'datafile', configurable.
The reason is this topobank issue: https://github.com/ContactEngineering/TopoBank/issues/579
Especially this comment: https://github.com/ContactEngineering/TopoBank/issues/579#issuecomment-817136579

But we can also later change this, when this issue is implemented in Topobank.

Reply:
I made this configurable. I had thought about a switch to tell the reader to use the NetCDF files, but this is better. In the future, I would actually default to using the NetCDF and not the original data files.

Reply:
(Since this will because a nested dictionary, we will have to update this code at a later time.)